protocol/src/main/java/org/geysermc/mcprotocollib/protocol/MinecraftProtocol.java

```diff
@@ -177,11 +177,11 @@ public PacketRegistry getPacketRegistry() {
         return this.stateRegistry;
     }
 
-    protected PacketEncryption enableEncryption(Key key) {
+    protected EncryptionConfig enableEncryption(Key key) {
         try {
-            return new AESEncryption(key);
+            return new EncryptionConfig(new AESEncryption(key));
         } catch (GeneralSecurityException e) {
-            throw new Error("Failed to enable protocol encryption.", e);
+            throw new IllegalStateException("Failed to enable protocol encryption.", e);
```

Comment:
would be nice to take this opportunity to rename the method and exception message from `enable` -> `create`